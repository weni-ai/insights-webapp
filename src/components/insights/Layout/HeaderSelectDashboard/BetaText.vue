<template>
  <section
    v-if="isConversationalDashboard || isHumanSupportDashboard"
    class="beta_text_container"
    data-testid="beta-container"
  >
    <p
      class="beta_text"
      data-testid="beta-text"
    >
      BETA
    </p>
  </section>
</template>

<script setup lang="ts">
import { useDashboards } from '@/store/modules/dashboards';
import { storeToRefs } from 'pinia';
import { computed } from 'vue';

const { currentDashboard } = storeToRefs(useDashboards());

const isConversationalDashboard = computed(() => {
  return currentDashboard.value.name === 'conversations_dashboard.title';
});

const isHumanSupportDashboard = computed(() => {
  return currentDashboard.value.name === 'human_support_dashboard.title';
});
</script>

<style lang="scss" scoped>
.beta_text_container {
  display: flex;
  padding: 0 $unnnic-spacing-nano;
  justify-content: center;
  align-items: center;
  gap: 10px;

  border-radius: $unnnic-border-radius-sm;
  background: $unnnic-color-weni-50;
}
.beta_text {
  color: $unnnic-color-weni-600;
  font-family: $unnnic-font-family-secondary;
  font-size: $unnnic-font-size-body-md;
  font-weight: $unnnic-font-weight-bold;
  line-height: $unnnic-line-height-md + $unnnic-font-size-body-md;
}
</style>
