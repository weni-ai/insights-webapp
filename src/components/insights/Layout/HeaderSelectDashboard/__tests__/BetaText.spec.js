import { beforeEach, describe, expect, it } from 'vitest';
import { mount } from '@vue/test-utils';

import BetaText from '../BetaText.vue';

describe('BetaText', () => {
  let wrapper;

  const createWrapper = () => mount(BetaText);

  const betaContainer = () => wrapper.find('[data-testid="beta-container"]');
  const betaText = () => wrapper.find('[data-testid="beta-text"]');

  beforeEach(() => {
    wrapper = createWrapper();
  });

  describe('Component rendering', () => {
    it('should render the beta container', () => {
      expect(betaContainer().exists()).toBe(true);
    });

    it('should render the beta text', () => {
      expect(betaText().exists()).toBe(true);
    });

    it('should display "BETA" text', () => {
      expect(betaText().text()).toBe('BETA');
    });
  });

  describe('Component structure', () => {
    it('should have correct CSS classes on container', () => {
      expect(betaContainer().classes()).toContain('beta_text_container');
    });

    it('should have correct CSS classes on text', () => {
      expect(betaText().classes()).toContain('beta_text');
    });

    it('should render as a section element', () => {
      expect(betaContainer().element.tagName).toBe('SECTION');
    });

    it('should render text as a paragraph element', () => {
      expect(betaText().element.tagName).toBe('P');
    });
  });

  describe('Snapshot', () => {
    it('should match snapshot', () => {
      expect(wrapper.element).toMatchSnapshot();
    });
  });
});
