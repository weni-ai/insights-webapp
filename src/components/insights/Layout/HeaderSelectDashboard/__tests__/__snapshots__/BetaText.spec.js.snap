// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`BetaText > Component structure > should match snapshot when not rendered 1`] = `<!--v-if-->`;

exports[`BetaText > Component structure > should match snapshot when rendered 1`] = `
<section
  class="beta_text_container"
  data-testid="beta-container"
  data-v-9ad7ff1d=""
>
  <p
    class="beta_text"
    data-testid="beta-text"
    data-v-9ad7ff1d=""
  >
     BETA 
  </p>
</section>
`;
